resources/design-document.md

```diff
@@ -0,0 +1,223 @@
+
+
+## 3. Use Cases
+
+U1. As a user, I want to create a new ad.
+
+U2. As a user, I want to view all my posted ad(s)
+
+U3. As a user, I want to update my ad(s) details. ***Can this be broken up into multiple cases - 
+
+U4. As a user, I want to be able to delete my ad(s).
+
+U5. As a user, I want to be able to update my bio details.
+
+U6. As a user, I want to be able to update my Name detail.
+
+U7. As a user, I want to be able to update my groups details.
+
+U8. As a user, I want to be able to update my roles details.
+
+U9. As a user, I want to bb able to view my individual ad(s).
+
+U10. As a user, I want to be able to update my ad's name.
+
+U11. As a user, I want to be able to update my ad's description.
+
+U12. As a user, I want to be able to update my ad's location.
+
+U13. As a user, I want to be able to update my ad's tags.
+
+U14. As a user, I want to be able to update my ad's salary.
+
+U15. As a user, I want to be able to update my ad's name.
+
+U16. As a user, I want to be able to view my user details.
+
+### 3.2  Stretch Use Cases
+
+U1. As a user, I want to be able to view other user's ads.
+
+U2. As a user, I want to be able to apply to another user's ad.
+
+U3. As a user, I want to be able to filter through other user's ads.
+
+U4. As a user, I want to be able to change bulletin boards.
+
+U5. As a user, I want to view venues associated with a board.
+
+## 4. Project Scope
+### 4.1 In Scope
+
+* Creating, retrieving, and updating an ad
+* Retrieving all ads a user has created
+* Updating a user's details. 
+
+### 4.2 Out of Scope
+
+* Multiple Boards (broader than Nashville area) - mid stretch
+* Venues as an entity rather than a String - closest stretch?
+* Allow users to apply to ads and view applications to their ads - farthest stretch 
+
+## 5. Proposed Architecture Overview
+
+The MLP will include creating, retrieving, and updating an ad. Included will also be the functionality to update a user's personal details.
+
+API Gateway and Lambda will be used to create these endpoints to satisfy our requirements:
+* `GetAd`
+* `CreateAd`
+* `UpdateAd`
+* `DeleteAd`
+* `GetUser`
+* `CreateUser`
+* `UpdateUser`
+
+Boards, users, ads, locationsm and venues will be stored within DynamoDB in their own respective tables. 
+
+
+## 6. API
+### 6.1 Public Models
+````
+// UserModel
+
+String userId;
+String name;
+String bio;
+List<String> groups;
+List<String> ads; //adId
+````
+
+````
+// AdModel
+
+String adId;
+String name;
+String description;
+Double salary; //double?
+String location; //venue name
+String userId;
+Set<String> tags;
+````
+
+````
+// LocationModel
+
+String locationId;
+String name;
+List<String> venues; //venueId
+````
+
+````
+// VenueModel
+
+String venueId;
+String name; 
+String description;
+String location;
+````
+
+````
+// BoardModel
+
+String boardId;
+String name;
+Set<String> users; //userIds
+Set<String> ads; //adIds
+String location;
+````
+
+### 6.2 Get Ad Endpoint
```

Comment:
Can you flesh out the endpoints with a little more detail? Take a look at the [example](https://github.com/nss-se-cohort-3/u5-ProjectTemplate/blob/main/resources/example-design-document.md?plain=1#L121) for ... examples. What does the endpoint do when invoked? Are there specific exceptions thrown if things don't work? Etc.